signer/signer_trust.go

```diff
@@ -85,3 +96,41 @@ func (trust *NotarySigner) GetKey(keyid string) data.PublicKey {
 	}
 	return data.NewPublicKey(data.KeyAlgorithm(publicKey.KeyInfo.Algorithm.Algorithm), publicKey.PublicKey)
 }
+
+// CheckHealth returns true if trust is healthy - that is if it can connect
+// to the trust server, and both the key management and signer services are
+// healthy
+func (trust *NotarySigner) CheckHealth(timeout time.Duration) error {
+	if e := trust.checkServiceHealth(
+		"Key Manager", trust.kmClient.CheckHealth, timeout); e != nil {
+		return e
+	}
+	return trust.checkServiceHealth(
+		"Signer", trust.sClient.CheckHealth, timeout)
+}
+
+type rpcHealthCheck func(
+	context.Context, *pb.Void, ...grpc.CallOption) (*pb.HealthStatus, error)
+
+// Generalized function that can check the health of both the signer client
+// and the key management client.
+func (trust *NotarySigner) checkServiceHealth(
+	serviceName string, check rpcHealthCheck, timeout time.Duration) error {
+
+	// Do not bother starting checking at all if the connection is broken.
+	if trust.clientConn.State() != grpc.Idle &&
+		trust.clientConn.State() != grpc.Ready {
+		return errors.New("Not currently connected to trust server.")
```

Comment:
Not important: if `fmt` is already imported, it's always an option to do a `fmt.Errorf` with no variable substitution, just saves importing `errors` for one-offs like this..